### ISSUE TITLE
Unable to specify thread binding via MPIEXEC_PREFLAGS due to excess CMake quoting

### ISSUE THREAD

**Issue by prckent** [QMCPACK/qmcpack#2326]
**Describe the bug**

OpenMPI will by default bind threads of MPI programs with a single task to a single core. This is always the wrong choice for QMCPACK (or any properly threaded application) and will always slow the tests. e.g.

```
export OMP_NUM_THREADS=128
mpirun -n 1 qmcpack input.xml
```
will not do what we want, despite being a reasonable invocation on, say, an AMD Rome processor with lots of cores.

Users have to specify "--bind-to none" for the desired behavior:
```
mpirun --bind-to none -n 1 input.xml
```

For our tests to use this flag, ideally we would use the MPIEXEC_PREFLAGS='--bind-to none' cmake setting, but when when invoked cmake put excess quotes, which confuses mpirun's parser.

```
mpirun "-n" "1" "--bind-to none" "input.xml"
```

The upshot of this bug ("feature") is that many of our tests take 16x longer to run than they should and we waste test hardware. 

Does anyone have a workaround to fix the quoting? This seems to be a known cmake problem but I could not find a workaround. I am specifically looking for a user invocable fix to the thread binding situation and not an edit to an OpenMPI configuration option. e.g. For the case where a user installed OpenMPI via spack or has no root access.

**To Reproduce**

Latest release or develop. Build and run the tests. Check top to see process mapping.

**Expected behavior**

Use the expected number of cores.

When I use --bind-to none by hand, I was able to get 12800% utilization on nitrogen.ornl.gov.

**System:**

OpenMPI 4.0.2, typical nightly build configurations. oxygen or nitrogen.

**Additional context**

**Comment by markdewing** [comment]
Try specifying as a CMake list - with semicolons separating the arguments

`MPIEXEC_PREFLAGS=--bind-to;none`

**Comment by PDoakORNL** [comment]
The shell quoting/parsing works that way by design, this is also how safe flag quoting works in pythons subprocess package without the unsafe shell flag for subprocesses set. I guess this is why most modern cli works with the -flag=value pattern. But the = at least doesn’t work with my openmpi mpirun.
 
The upshot is ‘--bind-to none‘ needs to be treated as two flags. The cmake flags are a list. And while usually you can fail to discriminate in Between space separated strings and lists in cmake you can’t here. To pass multiple list items to cmake in this sort of context list items are separated by ;
MPIEXEC_PREFLAGS='--bind-to;none'

**Comment by ye-luo** [comment]
Good to learn the semicolons syntax but I still have doubts why CMAKE_CXX_FLAGS can take spaces.

**Comment by PDoakORNL** [comment]
I agree it needs to go in manual. But you only need the ; because mpirun can only deal with argv[n]=“— bind-to” argv[n+1]=“none” and not argv[n]=“—bind-to none”

**Comment by prckent** [comment]
Thanks. The suggestion configures correctly, but does not execute as intended. While mpiexec does not give an error, it still seems to do the wrong thing in practice. I suspect that there are other ctest configuration options at play.

I also notice that the performance tests are invoked without mpiexec (!!), while the short tests are correctly invoked with mpiexec. Checking cdash I see that it is a problem we have had for some time that affects builds with Intel MPI as well. Probably we have several configuration issues to fix of our own.

Since I have limited time, I'll likely setup a build without MPI to use as a comparison. Unfortunately this can't test AFQMC, since it requires MPI.


When invoked by hand from build/tests/performance/NiO/dmc-a128-e1536-cpu, the following works correctly:

/home/pk7/apps/spack/opt/spack/linux-rhel8-zen2/gcc-9.2.0/openmpi-4.0.2-p7bprnejeuaqbwhyrqwid3cmqahcegv5/bin/mpiexec "-n" "1" "--bind-to" "none" "/home/pk7/projects/qmc/git_QMCPACK_qmcpack/build/bin/qmcpack" NiO-fcc-S32-dmc.xml

while

ctest -R "performance.*1536"

invoked from the build directory does not.

**Comment by ye-luo** [comment]
I found the root cause of not invoking MPIEXEC. Pushed my change to #2328.
ctest also needs -j to run parallel on many cores.
